src/include/column_inline.h

```diff
@@ -115,6 +140,13 @@ __col_insert_search_match(WT_INSERT_HEAD *ins_head, uint64_t recno)
     uint64_t ins_recno;
     int cmp, i;
 
+    /*
+     * CPU may reorder the reads and return a stale value of the tail.
+     *
+     * Place a read barrier here to avoid this issue.
+     */
+    WT_ORDERED_READ(ins, WT_SKIP_LAST(ins_head));
+
```

Comment:
It looks like you forgot to change line 151 to say `if (ins == NULL)` for this change.

Reply:
fixed